std/file.d

```diff
@@ -3200,6 +3200,219 @@ version(Posix) unittest //issue 11434
     assert(getAttributes(t2) == octal!100767);
 }
 
+/***************************************************
+Copy file or directory $(D from) to $(D to). When a directory is copied, its
+contents are copied as well, recursively.
+
+File timestamps are preserved, except for symbolic links.
+File attributes are preserved, if $(D preserve) equals $(D PreserveAttributes.yes).
+On Windows only $(D PreserveAttributes.yes) (the default on Windows) is supported.
+
+Existing target files are overwritten. Existing target directories are written
+into.
+
+With symbolic links, the links themselves are copied. That is, for every
+symbolic link in $(D from), a new link with the same target path is created in
+$(D to). Relative target paths are not changed. They may be invalidated by the
+copying, when they point outside of the copied directory tree.
+
+Special files (e.g. device files, FIFOs) cannot be copied with $(D copyRecurse).
+
+Params:
+    from = string or range of characters representing the existing
+        file/directory name
+    to = string or range of characters representing the target file/directory
+        name
+    preserve = whether to preserve the file attributes
+
+Throws: $(D FileException) on error.
+ */
+void copyRecurse(string from, string to,
+    PreserveAttributes preserve = preserveAttributesDefault)
+{
+    // Remove any trailing directory separators.
+    from = buildNormalizedPath(from);
+    to = buildNormalizedPath(to);
+
+    immutable attrs = getLinkAttributes(from);
+
+    if (attrIsFile(attrs)) copy(from, to, preserve);
```

Comment:
> And if I'm programming instead of using the command line, I want the same behavior.

That we've been using a scripting language (shell/batch) that uses the same ideas both for direct user interaction and automation is a gigantic historical mistake - just look at all the unintended `rm -rf /` happening in scripts because of missed escaping or unset variables.

> Anything else is just extra work that makes my life harder.

Sometimes making life harder is good, because if you make things too easy, it can make life VERY DIFFICULT when things go wrong.

By the same logic you would remove the `r` and `f` switches to `rm`, because they are redundant after all, aren't they?